mobile-browser-based-version/server/src/logic/federated/handlers.js

```diff
@@ -0,0 +1,547 @@
+import path from 'path';
+import fs from 'fs';
+import msgpack from 'msgpack-lite';
+import * as config from '../../../server.config.js';
+import {
+  averageWeights,
+  assignWeightsToModel,
+} from '../../helpers/tfjs_helpers.js';
+import tasks from '../../tasks/tasks.js';
+import * as tf from '@tensorflow/tfjs';
+import '@tensorflow/tfjs-node';
+
+const REQUEST_TYPES = Object.freeze({
+  CONNECT: 'connect',
+  DISCONNECT: 'disconnect',
+  SELECTION_STATUS: 'selection-status',
+  AGGREGATION_STATUS: 'aggregation-status',
+  POST_WEIGHTS: 'post-weights',
+  GET_WEIGHTS: 'get-weights',
+  POST_SAMPLES: 'post-samples',
+  GET_SAMPLES: 'get-samples',
+  GET_TASKS: 'get-tasks',
+});
+/**
+ * Fraction of client gradients required on final round of federated training
+ * to proceed to the pooling step.
```

Comment:
this is in every round, right? and pooling means aggregation here i think?

Reply:
exactly, pooling means aggregation and clients should this on the final epoch of a round only :)